source/simulator/helper_functions.cc

```diff
@@ -304,7 +304,13 @@ namespace aspect
 
         output_statistics();
 
-        if (parameters.run_postprocessors_on_initial_refinement)
+        // we only want to do the postprocessing here if it is not already done in
+        // the nonlinear iteration scheme
+        const bool run_postprocess = (!parameters.run_postprocessors_on_nonlinear_iterations
```

Comment:
Same applies here, if all schemes would be responsible if `run_postprocessors_on_nonlinear_iterations` is true, this would be significantly simpler.